src/degree_two_reduction.jl

```diff
@@ -1,13 +1,22 @@
-#NOTE: hardcoded for testing system 
+#NOTE: hardcoded for testing system
 function get_reduction(
-    A::IncidenceMatrix,
+    A::AdjacencyMatrix,
     sys::PSY.System,
     ::Val{NetworkReductionTypes.DEGREE_TWO},
 )
+    irreducible_buses = Set{Int}()
+    for c in get_components(StaticInjection, sys_DA)
+        push!(irreducible_buses, PSY.get_number(PSY.get_bus(c)))
+    end
+    irreducible_indices = Set([yb.lookup[2][i] for i in irreducible_buses])
+
+    arc_maps = find_degree2_chains(A.data, irreducible_indices)
```

Comment:
The complete logic isn't implemented yet to connect back the reduction data to the branches, but this code will find all the degree-2 reductions needed.